p384/src/arithmetic/hash2curve.rs

```diff
@@ -0,0 +1,148 @@
+        CtOption::new(self, 1.into())
+    }
+
+    fn is_torsion_free(&self) -> Choice {
+        1.into()
+    }
+}
+
+impl FromOkm for Scalar {
+    type Length = U72;
+
+    fn from_okm(data: &GenericArray<u8, Self::Length>) -> Self {
+        const F_2_288: Scalar = Scalar(U384::from_be_hex(
+            "000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000",
+        ));
+
+        let mut d0 = GenericArray::default();
+        d0[12..].copy_from_slice(&data[0..36]);
+        let d0 = Scalar(U384::from_be_byte_array(d0));
+
+        let mut d1 = GenericArray::default();
+        d1[12..].copy_from_slice(&data[36..]);
+        let d1 = Scalar(U384::from_be_byte_array(d1));
+
+        d0 * F_2_288 + d1
+    }
+}
+
+#[test]
+fn hash_to_scalar_voprf() {
+    use elliptic_curve::hash2curve::ExpandMsgXmd;
+    use hex_literal::hex;
+    use sha2::Sha384;
+
+    struct TestVector {
+        dst: &'static [u8],
+        seed: &'static [u8],
+        sk_sm: &'static [u8],
+    }
+
+    const TEST_VECTORS: &[TestVector] = &[
+        TestVector {
+            dst: b"HashToScalar-VOPRF08-\x00\x00\x04",
+            seed: &hex!("a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3"),
+            sk_sm: &hex!("b9ff42e68ef6f8eaa3b4d15d15ceb6f3f36b9dc332a3473d64840fc7b44626c6e70336bdecbe01d9c512b7e7d7e6af21"),
+        },
+        TestVector {
+            dst: b"HashToScalar-VOPRF08-\x01\x00\x04",
+            seed: &hex!("a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3"),
+            sk_sm: &hex!("42c4d1c15d27be015844404088967afe48c8ae96d4f00ce48e4d38ecabfb8feb5b748de625cdf81ab076745d6211be95"),
+        },
+    ];
+
+    for test_vector in TEST_VECTORS {
+        assert_eq!(
+            NistP384::hash_to_scalar::<ExpandMsgXmd<Sha384>>(&[test_vector.seed], test_vector.dst,)
+                .unwrap()
+                .to_bytes()
+                .as_slice(),
+            test_vector.sk_sm
+        );
+    }
+}
+
+#[test]
+fn from_okm_fuzz() {
+    use elliptic_curve::bigint::{Encoding, Limb, NonZero, UInt, U384};
+    use proptest::num::u64::ANY;
+    use proptest::{prelude::ProptestConfig, proptest};
+
+    fn simple_from_okm(data: GenericArray<u8, U72>) -> Scalar {
+        const LIMBS: usize = 72 * 8 / Limb::BIT_SIZE;
+        const MODULUS: NonZero<UInt<LIMBS>> = NonZero::from_uint(UInt::from_be_hex("000000000000000000000000000000000000000000000000ffffffffffffffffffffffffffffffffffffffffffffffffc7634d81f4372ddf581a0db248b0a77aecec196accc52973"));
+
+        let scalar = UInt::<LIMBS>::from_be_slice(&data) % MODULUS;
+
+        let mut out = [0; 72];
+
+        for (src, dst) in scalar
+            .into_limbs()
+            .into_iter()
+            .rev()
+            .zip(out.chunks_exact_mut(Limb::BYTE_SIZE))
+        {
+            dst.copy_from_slice(&src.to_be_bytes());
+        }
```

Comment:
This is just `ArrayEncoding::to_be_byte_array`, which isn't available for `U576`.

Reply:
I can potentially add `U576`

Reply:
As it turns out, `U576` is useful for `p521` as well, so I went ahead and added it:

https://github.com/RustCrypto/crypto-bigint/pull/94

Reply:
Released in `crypto-bigint` v0.4.5

Reply:
Amazing, that helped out a lot. How do I pull in 0.4.5 though?
- Update `elliptic-curve` to require `0.4.5`.
- Add a direct dependency to `0.4.5`.
- Only update `Cargo.lock`. That wouldn't work for downstream users, right?